### ISSUE TITLE
Unit of Depth measured..

### ISSUE THREAD

**Issue by abdur4373** [ialhashim/DenseDepth#32]
Hello @ialhashim  I wanted to ask about unit of depth measured, Is that in meters or mm...
You are clipping it between 10 and 1000, what  arre these two values...
In short I want the output of depth in meters(float), How could I get that...
Thanks

**Comment by ialhashim** [comment]
The original data for NYUv2 Depth is between 0 and 10 meters. The values I use in the code are arbitrary to that original depth but they should always be linearly related and convertible from and to it. Its up to you if you want to convert these values to any unit or simply normalize them from 0 to 1.

**Comment by abdur4373** [comment]
I would like to train your network on my own data set, I have few questions:
1) Should I use the raw depth images or do depth filling first using NYU MATLAB Toolbox?
2)Should I write some code or your train.py is ready o use code?
3)In case I want to train only few layers because my data set is very small but I want that It should perform very well in one environment(I do not need generalized results), How should I proceed with the training?
4)I have Dataset of about 1000 images, Will it work for training?
5)How many iterations will be needed approximately?


Thank you in advance...

**Comment by ialhashim** [comment]
1. Filling the missing depth values would be better if you intend to use the loss function we defined. Otherwise you may change the loss function to only consider filled values (by masking) and then use `l1` or `l2` metrics.
2. You will need to modify the data loader `data.py` and some small adjustment to ensure visualization and prediction codes are using the right range.
3. Just start with the relevant pre-trained models (NYU or KITTI) and then adjust the data loader to load your data and train.
4. Training from scratch on 1000 will give good training results, but validation an testing on other data depends on the generalization performance of the trained model. If you don't mind using NYU as pre-training you can start your training on the pretrained model. It should help the the model learn faster.
5. Again it depends on your target. For NYU we used which has 50,000 training images it typically needs around 12 to 24 hours to reach the results we show in the paper. Keep in mind that more iterations will result in over-fitting, i.e. worst generalization (real-world) performance.

In any case, all answers will be case dependent. Think about what you wan't to achieve and work on that data or testing scenario. In the end all current regression methods are still very low in quality (including ours). So manage your expectations, don't expect great success from the first try.

**Comment by abdur4373** [comment]
Thank you for your reply....
I am still stuck in the values your model is giving as an output to me...
`
return (np.clip(DepthNorm(predictions, maxDepth=1000), minDepth, maxDepth) / maxDepth)
`
This line  I am sue is just normalizing the output for display....
How can I get values in meters....?

**Comment by ialhashim** [comment]
Just multiply that by 10

**Comment by abdur4373** [comment]
Thank you for your reply, I want to train your model on my own data set, I have the NYU MATLAB processing toolbox In which format should I save the rgb and depth images and make zip file so that I can feed it directly to your model..... I am bothering you for so small issues ....sorry for that...
Thank again ....you are quite helping.... :+1:

**Comment by abdur4373** [comment]
And secondly should I change anything for training on almost 5000 images set...like freezing last layers.... I was working with pytorch and am new with tensorflow so thats why....

**Comment by abdur4373** [comment]
I multiplied it by 10 but I think that is not working for me I want depth value corresponding to every pixel in rgb??
1) Is there cropping involved or resizing once it is fed to the model and what is the output value of model(unit of that)..
because I think after using same
`
def DepthNorm(x, maxDepth):
    return maxDepth / x

def predict(model, images, minDepth=10, maxDepth=1000, batch_size=2):
    # Support multiple RGBs, one RGB image, even grayscale 
    if len(images.shape) < 3: images = np.stack((images,images,images), axis=2)
    if len(images.shape) < 4: images = images.reshape((1, images.shape[0], images.shape[1], images.shape[2]))
    # Compute predictions
    predictions = model.predict(images, batch_size=batch_size)
    predictions = predictions[0, :, :, 0]
    # Put in expected range
    return (np.clip(DepthNorm(predictions, maxDepth=1000), minDepth, maxDepth) / maxDepth)*10
`
code as yours you are dividing your values by max depth which gives inverse of depth I guess....
What I should do to get 240 *320 image of depth values in meters....

Thank you in advance....!!

**Comment by ialhashim** [comment]
For data format you can check the `nyu_data.zip` file for hints. I believe the RGBs are stored as JPG and the depths are as 8 bit PNG (loaded at this line https://github.com/ialhashim/DenseDepth/blob/cce1c10c0d21f3e3e9efb1a09b9af5f9880c2d8f/data.py#L70)

As for fine tuning for your own data that is up to you, I don't have much experience of how well it works. The `model.py` file has hints on how to do it here (https://github.com/ialhashim/DenseDepth/blob/cce1c10c0d21f3e3e9efb1a09b9af5f9880c2d8f/model.py#L26)

**Comment by ialhashim** [comment]
What are the ranges you are getting? I don't remember right now what the values are in (10 mm to 1000mm I think). Just look at a test example and you should see if they make sense in the real world (for example a small room should be in the range of 1 to 4 meters).

**Comment by abdur4373** [comment]
I am getting values like 170 as maximum and -3 as minimum, If your model is trained on nyu so how it is giving negative values before you do clipping and normalization...

**Comment by bat3a** [comment]
hi
about the Filling the missing depth value, is there a recommended aligorythm to use?
